convert/convert.go

```diff
@@ -155,51 +170,333 @@ func ConvertTSDBBlock(
 		encodingConcurrency: 1,
 		labelPageBufferSize: int(256 * units.KiB),
 		chunkPageBufferSize: int(2 * units.MiB),
+		writeConcurrency:    1,
 	}
 	for i := range opts {
 		opts[i](cfg)
 	}
 	start, end := day.MinT(), day.MaxT()
 	name := schema.BlockNameForDay(day)
-	rr, err := newIndexRowReader(ctx, start, end, blks, indexReaderOpts{
-		sortLabels:  cfg.sortLabels,
-		concurrency: cfg.encodingConcurrency,
-	})
+
+	shardedRowReaders, err := shardedIndexRowReader(ctx, start, end, blks, *cfg)
 	if err != nil {
-		return fmt.Errorf("unable to create index rowreader: %w", err)
-	}
-	defer errcapture.Do(&rerr, rr.Close, "index row reader close")
-
-	converter := newConverter(
-		name,
-		start,
-		end,
-		rr,
-		bkt,
-		cfg.rowGroupSize,
-		cfg.numRowGroups,
-		cfg.buildSortingColumns(),
-		cfg.buildBloomfilterColumns(),
-		cfg.labelBufferPool,
-		cfg.chunkbufferPool,
-		cfg.labelPageBufferSize,
-		cfg.chunkPageBufferSize,
-	)
+		return 0, fmt.Errorf("failed to create sharded TSDB row readers: %w", err)
+	}
+	defer func() {
+		for _, rr := range shardedRowReaders {
+			_ = rr.Close()
+		}
+	}()
+	logger.Info("starting parallel block conversion", "shards", len(shardedRowReaders), "write_concurrency", cfg.writeConcurrency)
```

Comment:
I think thats the only line we log really, that could easily live outside and in the main function for converter. Can we drop the logger from the function signature maybe?